libkbfs/bserver_remote.go

```diff
@@ -27,11 +26,18 @@ const (
 	BServerTokenExpireIn = 2 * 60 * 60 // 2 hours
 )
 
+type blockServerRemoteConfig interface {
+	diskBlockCacheGetter
+	codecGetter
+	signerGetter
+	currentInfoGetterGetter
+	logMaker
+}
+
 // BlockServerRemote implements the BlockServer interface and
 // represents a remote KBFS block server.
 type BlockServerRemote struct {
-	codec      kbfscodec.Codec
-	cig        currentInfoGetter
+	config     blockServerRemoteConfig
```

Comment:
I disagree. Staying with a consistent pattern is really awesome, because then suddenly we can pass around our regular `Config` implementation and components specify and pick out the methods they need. It also makes implementing test interfaces much easier.